### PR TITLE
Feature/create itinerary preview uses real created itinerary values

### PR DESCRIPTION
**Pull request by gunnm0**
This is a small commit where I made CreateItineraryPreview use real values of the itinerary being created instead of hard coded values. 
I also added the fact that a title can't be longer than 80 characters, done and tested.

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt [hunk 1]
```diff
@@ -29,6 +29,8 @@ object TestTags {
   const val CREATION_SCREEN_NAV_BAR = "Creation Screen Top navigation bar"
   const val CREATION_FINISH_BUTTON = "Preview Finish Itinerary Creation Button"
 
+  const val INVALID_INPUT = "invalid input"
+
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt:33]: What's up with the whitespace?

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt [hunk 2]
```diff
@@ -44,7 +47,14 @@ fun CreationStepChooseDescriptionScreen(createItineraryViewModel: CreateItinerar
               value = title,
               onValueChange = {
                 title = it
-                createItineraryViewModel.getNewItinerary()?.title(it)
+                validTitle =
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt:50]: Could you explain what validTitle means?

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt [hunk 3]
```diff
@@ -16,8 +16,13 @@ class CreateItineraryViewModel(
     ItineraryViewModel(
         itineraryRepository, directionsRepository, locationsRepository, locationClient) {
   /** New itinerary that the signed in user is currently building */
+  private val maxTitleLength = 80
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt:19]: little weird with how this line is placed given that the comment above of it was not intended for this line, but for the one beneath.

just a quirk though

**Review by ethangraham2001 — CHANGES_REQUESTED** [review]
a few small changes

File: app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt [hunk 4]
```diff
@@ -29,6 +29,8 @@ object TestTags {
   const val CREATION_SCREEN_NAV_BAR = "Creation Screen Top navigation bar"
   const val CREATION_FINISH_BUTTON = "Preview Finish Itinerary Creation Button"
 
+  const val INVALID_INPUT = "invalid input"
```
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt:32]: maybe, to keep coherence, you can change "invalid" to "Invalid"

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt:1]: Lgtm, clean code

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewBanner.kt [hunk 5]
```diff
@@ -35,7 +35,7 @@ fun CreationStepPreviewBanner(
           true,
       )
 
-  // set the itineary, not permanent, will do later when logic is applied
+  // set the itinerary, not permanent, will do later when logic is applied
   val itinerary = FakeItinerary.SWITZERLAND
```
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewBanner.kt:39]: Needs to be changed with new implementation

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt (hunk 3 above)
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt:19]: Magic value ? Maybe you can add a constant to get a cleaner code.

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt [hunk 6]
```diff
@@ -60,6 +65,15 @@ class CreateItineraryViewModel(
     newItineraryBuilder?.description = description
   }
 
+  /** @returns true if the title is valid * */
+  fun validTitle(title: String): Boolean {
+    return title.length < maxTitleLength
+  }
+
+  fun invalidTitleMessage(): String {
+    return "title field must be shorter than ${maxTitleLength} characters"
```
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt:74]: Maybe you can add some capital letters and a dot at the end of a sentence when you write some text, i think it's cleaner.

File: app/src/test/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescriptionKtTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescriptionKtTest.kt:1]: Clean code, well done!

File: app/src/test/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewKtTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewKtTest.kt:1]: Good tests!

**Review by engjellismaili — COMMENTED** [review]
Overall clean code, maybe some small fix to do, but not too important.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt (hunk 1 above)
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt:33]: style

Comment (ethangraham2001) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt:33]: ok

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt (hunk 2 above)
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt:50]: the title should be valid, for now it must be less than 80 characters but in the future we could imagine other filters such as the n-word.

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt (hunk 3 above)
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt]: changed

File: app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt (hunk 4 above)
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt:32]: ok

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepChooseDescription.kt:1]: ty

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt (hunk 3 above)
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt]: changed

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt (hunk 6 above)
Comment (gunnm0) [reply app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt]: yes but this is nicer for an error message

**Pull request merged**